packages/x-data-grid-premium/src/tests/rowGrouping.DataGridPremium.test.tsx

```diff
@@ -2381,6 +2382,45 @@ describe('<DataGridPremium /> - Row grouping', () => {
         // Corresponds to rows id 0, 1, 2 because of Cat A, ann id 4 because of Cat 1
         expect(getColumnValues(1)).to.deep.equal(['', '0', '1', '2', '', '4']);
       });
+
+      it('should keep the correct count of the children and descendants in the filter state', () => {
+        const extendedColumns = [
+          ...baselineProps.columns,
+          {
+            field: 'value1',
+          },
+        ];
+
+        const extendedRows = rows.map((row, index) => ({ ...row, value1: `Value${index}` }));
+        const additionalRows = [
+          { id: 5, category1: 'Cat A', category2: 'Cat 2', value1: 'Value5' },
+          { id: 6, category1: 'Cat A', category2: 'Cat 2', value1: 'Value6' },
+          { id: 7, category1: 'Cat B', category2: 'Cat 1', value1: 'Value7' },
+        ];
+
+        render(
+          <Test
+            columns={extendedColumns}
+            rows={[...extendedRows, ...additionalRows]}
+            initialState={{ rowGrouping: { model: ['category1', 'category2'] } }}
+            defaultGroupingExpansionDepth={3}
+            rowGroupingColumnMode="multiple"
+          />,
+        );
+
+        const { filteredChildrenCountLookup, filteredDescendantCountLookup } =
+          apiRef.current.state.filter;
+
+        expect(filteredChildrenCountLookup['auto-generated-row-category1/Cat A']).to.equal(2);
```

Comment:
Should we test `aria-posinset` and `aria-setsize` here instead of the internal state?

Reply:
this one actually tests new `filteredChildrenCountLookup`
aria attributes are tested further below